python/setup.py

```diff
@@ -29,28 +40,130 @@
   'win32',
 ]
 
-cwd = os.path.abspath(os.path.dirname(__file__))
-with open(cwd + '/README.rst') as f:
-  readme = f.read()
-
-with open(cwd + '/requirements.txt') as f:
-  INSTALL_REQUIRES = [i.strip() for i in f.readlines()]
-
-with open(cwd + '/test_requirements.txt') as f:
-  TEST_REQUIRES = [i.strip() for i in f.readlines()]
-
-setup(name='sbp',
-      description='Python bindings for Swift Binary Protocol',
-      long_description=readme,
-      use_scm_version = {'root': '..', 'relative_to': __file__},
-      setup_requires=['setuptools_scm'],
-      author='Swift Navigation',
-      author_email='dev@swiftnav.com',
-      url='https://github.com/swift-nav/libsbp',
-      classifiers=CLASSIFIERS,
-      packages=PACKAGES,
-      platforms=PLATFORMS,
-      install_requires=INSTALL_REQUIRES,
-      tests_require=TEST_REQUIRES,
-      use_2to3=False,
-      zip_safe=False)
+MAJOR = 2
+MINOR = 4
+MICRO = 8
+
+IS_RELEASED = False
+
+VERSION = '%d.%d.%d' % (MAJOR, MINOR, MICRO)
+
+VERSION_PY_TEMPLATE = """\
+# THIS FILE IS GENERATED FROM TRAITS SETUP.PY
+version = '{version}'
+full_version = '{full_version}'
+git_revision = '{git_revision}'
+is_released = {is_released}
+if not is_released:
+    version = full_version
+"""
+
+VERSION_PY_PATH = 'sbp/_version.py'
+
+
+# Return the git revision as a string
+def git_version():
+    def _minimal_ext_cmd(cmd):
+        # construct minimal environment
+        env = {}
+        for k in ['SYSTEMROOT', 'PATH']:
+            v = os.environ.get(k)
+            if v is not None:
+                env[k] = v
+        # LANGUAGE is used on win32
+        env['LANGUAGE'] = 'C'
+        env['LANG'] = 'C'
+        env['LC_ALL'] = 'C'
+        out = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, env=env,
+        ).communicate()[0]
+        return out
+    try:
+        out = _minimal_ext_cmd(['git', 'describe', '--tags'])
```

Comment:
what will happen if someone pip installs sbp WITHOUT git in their path?

Reply:
It should resolve to "unknown", see `if match is None` below.

Reply:
Thus, if it was an unreleased tarball of libsbp, then it would show as "v2.4.9.dev0+gUnknown" as it's version.